scenario/helpers.go

```diff
@@ -95,3 +100,83 @@ func Contains(list []string, match func(s string) bool) bool {
 	}
 	return false
 }
+
+// IndexOf returns index of first match in stringSlice or else -1
+func IndexOf(match string, stringSlice []string) (int, bool) {
+	for i, str := range stringSlice {
+		if str == match {
+			return i, true
+		}
+	}
+	return -1, false
+}
+
+// TODO move enum related code to its own package
+type (
+	MutableEnum interface {
```

Comment:
Why do we need these new types?

Reply:
Fewer lines of boilerplate when implementing json.Marshaler and json.Unmarshaler for Enum-types. Had to do that for two Enum-types at the end of clickactionbutton.go.

Reply:
What's the big difference from existing
func (em *EnumMap) UnMarshal(arg []byte) (int, error) {...}
func (em *EnumMap) StringDefault(i int, dflt string) string {...}